src/azure-cli/azure/cli/command_modules/iot/tests/hybrid_2020_09_01/_test_utils.py

```diff
@@ -3,78 +3,90 @@
 # Licensed under the MIT License. See License.txt in the project root for license information.
 # --------------------------------------------------------------------------------------------
 
+import datetime
 from os.path import exists
 import os
-from OpenSSL import crypto
+from cryptography import x509
+from cryptography.x509.oid import NameOID
+from cryptography.hazmat.primitives import serialization, hashes
+from cryptography.hazmat.primitives.asymmetric import rsa
+from cryptography.hazmat.primitives.serialization import load_pem_private_key
 
 
 def _create_test_cert(cert_file, key_file, subject, valid_days, serial_number):
     # create a key pair
-    k = crypto.PKey()
-    k.generate_key(crypto.TYPE_RSA, 2046)
-
-    # create a self-signed cert with some basic constraints
-    cert = crypto.X509()
-    cert.get_subject().CN = subject
-    cert.gmtime_adj_notBefore(-1 * 24 * 60 * 60)
-    cert.gmtime_adj_notAfter(valid_days * 24 * 60 * 60)
-    cert.set_version(2)
-    cert.set_serial_number(serial_number)
-    cert.add_extensions([
-        crypto.X509Extension(b"basicConstraints", True, b"CA:TRUE, pathlen:1"),
-        crypto.X509Extension(b"subjectKeyIdentifier", False, b"hash",
-                             subject=cert),
-    ])
-    cert.add_extensions([
-        crypto.X509Extension(b"authorityKeyIdentifier", False, b"keyid:always",
-                             issuer=cert)
-    ])
-    cert.set_issuer(cert.get_subject())
-    cert.set_pubkey(k)
-    cert.sign(k, 'sha256')
-
-    cert_str = crypto.dump_certificate(crypto.FILETYPE_PEM, cert).decode('ascii')
-    key_str = crypto.dump_privatekey(crypto.FILETYPE_PEM, k).decode('ascii')
-
-    open(cert_file, 'w').write(cert_str)
-    open(key_file, 'w').write(key_str)
+    key = rsa.generate_private_key(public_exponent=65537, key_size=2048)
+
+    # create a self-signed cert
+    subject_name = x509.Name(
+        [
+            x509.NameAttribute(NameOID.COMMON_NAME, subject),
+        ]
+    )
+    cert = (
+        x509.CertificateBuilder()
+        .subject_name(subject_name)
+        .issuer_name(subject_name)
+        .public_key(key.public_key())
+        .serial_number(serial_number)
+        .not_valid_before(datetime.datetime.utcnow())
+        .not_valid_after(
+            datetime.datetime.utcnow() + datetime.timedelta(days=valid_days)
+        )
+        .sign(key, hashes.SHA256())
+    )
+
+    key_dump = key.private_bytes(
+        encoding=serialization.Encoding.PEM,
+        format=serialization.PrivateFormat.TraditionalOpenSSL,
+        encryption_algorithm=serialization.NoEncryption(),
+    ).decode("utf-8")
+    cert_dump = cert.public_bytes(serialization.Encoding.PEM).decode("utf-8")
```

Comment:
Maybe this procedure can be extracted to a common function. 😉

Reply:
added to backlog